src/Graphics/SpriteBatch.cs

```diff
@@ -144,6 +147,8 @@ public SpriteBatch(GraphicsDevice graphicsDevice)
 
 			vertexInfo = new VertexPositionColorTexture4[MAX_SPRITES];
 			textureInfo = new Texture2D[MAX_SPRITES];
+			spriteInfos = new SpriteInfo[MAX_SPRITES];
```

Comment:
Just realized that this would have to be either pinned or AllocHGlobal'd, otherwise the GC could move the location of the array around mid-batch and cause all the sortedSprite pointers to be the wrong location and crash.

Reply:
In this case I was referring to `spriteInfos` - for example:

1. spriteInfos allocated, memory location is `0xC000`
2. spriteBatch.Begin();
3. spriteBatch.Draw(...) called 5 times, pointers are `0xC000` through, say, `0xC040`
4. GC acts, decided to move spriteInfos, memory location is now `0xF000`
5. spriteBatch.Draw(...) called once, pointer is `0xC050`
6. spriteBatch.End() called, sorts sprites
7. Sorter tries to access `0xC000`, crash!

So spriteInfos has to be persistently-pinned, or it has to be allocated off the GC heap entirely.

Reply:
Bingo, thanks Ethan. I hate garbage collection.

Rather than go to the point of allocating from a different heap, I moved things closer together so they could be pinned together during the critical transformation.  Maybe it will use the cache better too overall.

Running the same stats (though different day so who knows what my PC is juggling) I get...
```

Technique: Fna
           Type Average     Min     Max  Median
       Deferred    3816    2367    6771    4569
      Immediate   34637   30988   55296   43142
        Texture    7445    5122   11383    8253
    BackToFront    9401    6440   19293   12867
    FrontToBack   10202    6413   16153   11283
```

Looks unchanged performance wise.